.bit/responses/3.2-Week 3 Step 2.md

```diff
@@ -1,80 +1,262 @@
+
+```jsx
+async function loadFile(event){
+    console.log("Got picture!");
+    var image = document.getElementById("output");
+    // Get image from output 
+    image.src = URL.createObjectURL(event.target.files[0])
+    // load inputted image into the image src and display
+}
+```
+
+Then, `handle()` is called when the file is submitted to POST the image and username. The image is sent in the body, and username is sent as a header. *Lines 15-30*
+
+`requestJS.js`:
+
+```js
+async function handle(event) {
+    event.preventDefault();
+    document.getElementById("output").src = "img/logo.png";
+    console.log("Loading picture");
+    var username = document.getElementById("username").value;
+    if (username.includes(' ') == true || username == '') {
+      alert("Invalid username. A username cannot contain a space.");
+      window.location.reload();
+      return;
+    }
+
+    document.getElementById("upload").src = "img/doneupload.gif";
+    $('#submit').html(`Thank you for your image, use "${username}" to receive your pdf.`);
+
+    var myform = document.getElementById("image-form");
+        var payload = new FormData(myform);
+
+        console.log("Posting your image...");
+        const resp = await fetch("https://bunnimage1.azurewebsites.net/api/uploadTrigger?code=Ia/3vNYiimrORzxaZRGlmdm695dnnSpCP0qd7R1k1WaUeRO2JUfGtg==", {
+            method: 'POST',
+            headers: {
+                'username' : username
+            },
+            body: payload
+        });
+
+        var data = await resp.json();
+        console.log(data);
+        console.log("Blob has been stored successfully!")
+}
+```
+
+> Be sure to change the function url on Line 19 to the one of your upload image Function!
+
+![https://user-images.githubusercontent.com/69332964/99188529-73369a00-272a-11eb-93df-04fdce5381df.png](https://user-images.githubusercontent.com/69332964/99188529-73369a00-272a-11eb-93df-04fdce5381df.png)
+
+### Deploy your code
+
+There are three options:
+
+- Try doing it locally with the **live server extension** for VS Code
+- Try [Azure Web Apps](https://azure.microsoft.com/en-us/services/app-service/web/)
+- Try [repl.it](https://repl.it/languages/html)
+
+### Update CORS Settings
+
+> This is a crucial step!! 😱 If you don't change your CORS (Cross-origin resource sharing) settings, the POST request won't work. This tells the Function App what domains can access our Azure Function.
+
+**Options:**
+
+- **Recommended**: Change it to a wildcard operator (`*`), which allows *all* origin domains to make requests
+    - Be sure to remove any other existing inputs before attempting to save with wildcard
+
+        ![https://user-images.githubusercontent.com/69332964/99188905-6f0b7c00-272c-11eb-8142-f91882227c78.png](https://user-images.githubusercontent.com/69332964/99188905-6f0b7c00-272c-11eb-8142-f91882227c78.png)
+
+- Change it to the domain you are using to host your code
+
+### Home stretch! 🏃🏻‍♀️

```

Comment:
![image](https://user-images.githubusercontent.com/69332964/120042049-63e8b100-bfd7-11eb-9290-810a307505af.png)

Instead of this part below, you'll need to add a "test your work" that's similar to what's above ^^ that shows students how to attach an image to Postman and explains to them what output they should get